terraform/roles.tf

```diff
@@ -288,3 +288,46 @@ resource aws_iam_role_policy_attachment ec2_readonly {
   role       = aws_iam_role.caascad_operator.name
   policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess"
 }
+
+resource aws_iam_role lambda-promtail {
+  name = "caascad-lambda-promtail-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [{
+      Action = "sts:AssumeRole"
+      Effect = "Allow"
+      Principal = {
+        Service = "lambda.amazonaws.com"
+      }
+    }]
+  })
+
+  tags = {
+    caascad = "true"
+  }
+}
+
+data aws_iam_policy_document lambda-promtail {
+  statement {
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+    resources = [
+      "arn:aws:logs:*:*:*",
+    ]
+  }
+}
+
+resource aws_iam_policy lambda-promtail {
+  name = "caascad-lambda-promtail-policy"
```

Comment:
Although the current use-case is for lamda promtail the policy is more generic than that and the name should reflect that

Reply:
Policy was renamed `caascad-log-group-write-policy`